### DISCUSSION TITLE
Add `pac plugin push --pluginId <uniquename>`

### DISCUSSION THREAD

**Discussion by filcole** [microsoft/powerplatform-build-tools#930] [Ideas]
We can update a plugin package via 

```
pac plugin push  --pluginId 532b974f-e827-ef11-840a-002248c60dce
```

However we have to find the GUID via the Plugin Registration Tool (`pac tool prt`), see screenshot below:

![image](https://github.com/microsoft/powerplatform-build-tools/assets/6078398/7e4a57ff-f665-4be4-9793-349325cc38ad)

It would be more user-friendly to be able to specify the unique name. i.e. something similar to below.

```powershell
pac plugin push --pluginid xxx_Plugins
```

Note: The xxx_Plugins just an example name.

**Comment by devkeydet** [comment]
Agree that it would be more convenient to not have to lookup the guid.  In the interim, here's what I've been doing to get the guid without having to launch another tool...

`pac pfx repl`

At the prompt, type:

`Filter('Plug-in Assemblies', Name = "MyAssemblyName")`

Example:
![image](https://github.com/microsoft/powerplatform-build-tools/assets/4499296/9c188d80-0ee5-490a-9dd2-acfbd72ed04c)

**Comment by parvezghumra** [comment]
I'm wondering why `pac plugin push` is limited to only _**updating existing**_ plugins (by virtue of the `--pluginid` argument being mandatory?

Could this not be enhanced to allow new plugins/plugin packages to be registered too (by making `--pluginid` optional?

Furthermore could we get also additional commands to also manage (including all possible operations(eg. Create/Update/Delete/Disable/Enable) plugin steps, images, service endpoints, webhooks, custom APIs, data providers etc.

Apologies for hijacking the request :)
